src/common/adbc/adbc.cpp

```diff
@@ -594,15 +839,21 @@ AdbcStatusCode StatementPrepare(struct AdbcStatement *statement, struct AdbcErro
 	if (status != ADBC_STATUS_OK) {
 		return status;
 	}
-
 	return ADBC_STATUS_OK;
 }
 
 AdbcStatusCode StatementSetSqlQuery(struct AdbcStatement *statement, const char *query, struct AdbcError *error) {
+	if (!error) {
+		return ADBC_STATUS_UNKNOWN;
+	}
```

Comment:
That might make more sense yea, less duplication

currently we do
```c++
	if (!error) {
		return ADBC_STATUS_INVALID_ARGUMENT;
	}
``` in SetErrorMaybe

Reply:
Hmm looking at the docs of the errors, it doesn't look like this is what it's for
```c++
/// \brief An unknown error occurred.
///
/// May indicate a driver-side or database-side error.
#define ADBC_STATUS_UNKNOWN 1
```

Reply:
I ended up removing SetErrorMaybe, since it still returns an AdbcStatusCode, you still need to check it anyways

So `if (condition) { SetError(error, message); return error; }` is cleaner in my opinion.
I really think it might not hurt to make a macro (or multiple) for this instead, `ReturnIfError(condition, error, message, error type);`